proxy/proxy.php

```diff
@@ -127,18 +153,34 @@ function civiproxy_redirect4($url_requested, $parameters, $credentials) {
   curl_setopt($curlSession, CURLOPT_RETURNTRANSFER,1);
   curl_setopt($curlSession, CURLOPT_TIMEOUT, 30);
   curl_setopt($curlSession, CURLOPT_SSL_VERIFYHOST, 2);
+  curl_setopt($curlSession, CURLOPT_FAILONERROR, true);
   if (!empty($target_interface)) {
     curl_setopt($curlSession, CURLOPT_INTERFACE, $target_interface);
   }
   if (file_exists(dirname(__FILE__).'/target.pem')) {
     curl_setopt($curlSession, CURLOPT_CAINFO, dirname(__FILE__).'/target.pem');
   }
 
+  $prepareRedirect = new prepareRedirect($url, $parameters, $curlSession, 3);
+  CiviProxy::dispatchEvent($prepareRedirect);
+
+  if ($prepareRedirect->hasRepsone) {
+    // handle headers - simply re-outputing them
+    foreach ($prepareRedirect->responseHeaders as $header_line){
+      header(trim($header_line));
+    }
+    echo $prepareRedirect->repsone;
+    curl_close ($curlSession);
+    exit();
+  }
+
   //Send the request and store the result in an array
   $response = curl_exec($curlSession);
 
   // Check that a connection was made
   if (curl_error($curlSession)){
+    $redirectError = new redirectError($curlSession, 4);
+    CiviProxy::dispatchEvent($redirectError);
     civiproxy_http_error(curl_error($curlSession), curl_errno($curlSession));
 
   } else {
```

Comment:
This is missing the `finishRedirect` event being dispatched, as is done in the APIv3 implementation above.

Reply:
A good spotted.